README.md

```diff
@@ -1,12 +1,85 @@
-# mysql-operator
+# MySQL Operator
 
 ## Description
 
-TODO: Describe your charm in a few paragraphs of Markdown
+The [MySQL](https://www.mysql.com/) operator provides an open-source relational database management system (RDBMS). This repository contains a Juju Charm for deploying MySQL on Kubernetes clusters.
+
+
+## Setup
+
+A typical setup using [snaps](https://snapcraft.io/), for deployments to a [microk8s](https://microk8s.io/) cluster can be done using the following commands
+
+```bash
+$ sudo snap install microk8s --classic
+$ microk8s.enable dns storage registry dashboard
+$ sudo snap install juju --classic
+$ juju bootstrap microk8s microk8s
+$ juju create-storage-pool operator-storage kubernetes storage-class=microk8s-hostpath
+```
+
+## Build
+
+Install the charmcraft tool
+
+```bash
+$ sudo snap install charmcraft
+```
+
+Build the charm in this git repository
+
+```bash
+$ charmcraft build
+```
 
 ## Usage
 
-TODO: Provide high-level usage, such as required config or relations
+
+Create a Juju model for your operators, say "mysql"
+
+```bash
+$ juju add-model mysql
+```
+
+Deploy a single unit of MySQL using its default configuration
+
+```bash
+$ juju deploy ./mysql.charm --resource mysql-image=ubuntu/mysql:lates
+```
+
+As in this example we did not especify the password for the root user, the charm will generate one, and you can get it watching the juju debug-log:
+
+
+```bash
+$ juju debug-log -l
+
+...
+
+application-mysql: 13:45:16 WARNING unit.mysql/0.juju-log The randomly generated MYSQL_ROOT_PASSWORD is: T4NgXUeXYlJBBD37icuM
+application-mysql: 13:45:16 WARNING unit.mysql/0.juju-log Please change it as soon as possible!
+```
+
+If required, yo can remove the deployed completely:
```

Comment:
grammar: "deployment" not "deployed". Also spelling "you" not "yo"

Reply:
Fixed!